ui/packages/consul-ui/app/components/consul/kind/index.js

```diff
@@ -1,5 +1,19 @@
 import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { titleize } from 'ember-cli-string-helpers/helpers/titleize';
+import { humanize } from 'ember-cli-string-helpers/helpers/humanize';
+
+const normalizedGatewayLabels = {
+  'api-gateway': 'API Gateway',
+  'mesh-gateway': 'Mesh Gateway',
+  'ingress-gateway': 'Ingress Gateway',
+  'terminating-gateway': 'Terminating Gateway',
+};
 
 export default Component.extend({
   tagName: '',
+  Name: computed('item.Kind', function () {
+    const name = normalizedGatewayLabels[this.item.Kind];
+    return name ? name : titleize(humanize(this.item.Kind));
+  }),
```

Comment:
This works great as is, but if you'd like to update this component to use the newer ember (Octane) conventions for component you can update this to be:

```
import Component from '@glimmer/component';
import { titleize } from 'ember-cli-string-helpers/helpers/titleize';
import { humanize } from 'ember-cli-string-helpers/helpers/humanize';

const normalizedGatewayLabels = {
  'api-gateway': 'API Gateway',
  'mesh-gateway': 'Mesh Gateway',
  'ingress-gateway': 'Ingress Gateway',
  'terminating-gateway': 'Terminating Gateway',
};

export default class CounterComponent extends Component {
  get name() {
    const name = normalizedGatewayLabels[this.args.item.Kind];
    return name ? name : titleize(humanize(this.args.item.Kind));
  }
}
```

However, doing this will also require updating the associated template's usages of `item` and `withInfo` to be `@item` and `@withInfo` to denote the usage of passed in arguments.

Definitely not required, just providing it as an option.

https://guides.emberjs.com/release/components/component-arguments-and-html-attributes/#toc_arguments
and
https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/#component-properties__tracked-vs-cp

Reply:
Ooh, neat to know, that'd at least let us start removing some of the deprecation warnings that get thrown.

I'm thinking the bigger change can probably wait since we'll likely have some more changes going into this component in the next release i.e. at minimum, a learn tutorial -- and since we're trying to merge for a patch release, I don't want to make too many changes (especially not having touched Ember code for ~8 years).